pkg/webhook/servicecatalog/serviceclass/mutation/handler.go

```diff
@@ -0,0 +1,99 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mutation
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+
+	sc "github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog/v1beta1"
+	webhookutil "github.com/kubernetes-incubator/service-catalog/pkg/webhookutil"
+
+	admissionTypes "k8s.io/api/admission/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+// CreateUpdateHandler handles ServiceClass
+type CreateUpdateHandler struct {
+	decoder *admission.Decoder
+}
+
+var _ admission.Handler = &CreateUpdateHandler{}
+
+// Handle handles admission requests.
+func (h *CreateUpdateHandler) Handle(ctx context.Context, req admission.Request) admission.Response {
+	traced := webhookutil.NewTracedLogger(req.UID)
+	traced.Infof("Start handling operation: %s for %s: %q", req.Operation, req.Kind.Kind, req.Name)
+
+	cb := &sc.ServiceClass{}
+	if err := webhookutil.MatchKinds(cb, req.Kind); err != nil {
+		traced.Errorf("Error matching kinds: %v", err)
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	if err := h.decoder.Decode(req, cb); err != nil {
+		traced.Errorf("Could not decode request object: %v", err)
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	mutated := cb.DeepCopy()
+	switch req.Operation {
+	case admissionTypes.Create:
+		h.mutateOnCreate(ctx, mutated)
+	case admissionTypes.Update:
+		h.mutateOnUpdate(ctx, mutated)
+	default:
+		traced.Infof("ServiceClass mutation wehbook does not support action %q", req.Operation)
+		return admission.Allowed("action not taken")
+	}
+	h.syncLabels(mutated)
+	rawMutated, err := json.Marshal(mutated)
+	if err != nil {
+		traced.Errorf("Error marshaling mutated object: %v", err)
+		return admission.Errored(http.StatusInternalServerError, err)
+	}
+
+	traced.Infof("Completed successfully operation: %s for %s: %q", req.Operation, req.Kind.Kind, req.Name)
+	return admission.PatchResponseFromRaw(req.Object.Raw, rawMutated)
+}
+
+var _ admission.DecoderInjector = &CreateUpdateHandler{}
+
+// InjectDecoder injects the decoder
+func (h *CreateUpdateHandler) InjectDecoder(d *admission.Decoder) error {
+	h.decoder = d
+	return nil
+}
+
+func (h *CreateUpdateHandler) mutateOnCreate(ctx context.Context, binding *sc.ServiceClass) {
+
+}
+
+func (h *CreateUpdateHandler) mutateOnUpdate(ctx context.Context, obj *sc.ServiceClass) {
+	// TODO: implement logic from pkg/registry/servicecatalog/binding/strategy.go
+}
+
+func (h *CreateUpdateHandler) syncLabels(obj *sc.ServiceClass) {
+	if obj.Labels == nil {
+		obj.Labels = make(map[string]string)
+	}
+
+	obj.Labels[sc.GroupName+"/spec.externalID"] = obj.Spec.ExternalID
+	obj.Labels[sc.GroupName+"/spec.externalName"] = obj.Spec.ExternalName
+	obj.Labels[sc.GroupName+"/spec.clusterServiceBrokerName"] = obj.Spec.ServiceBrokerName
```

Comment:
```suggestion
	obj.Labels[sc.GroupName+v1beta1.FilterSpecServiceBrokerName] = obj.Spec.ServiceBrokerName
```

in below comment you will find more information